demos/microcloud-deploy/main.tf

```diff
@@ -0,0 +1,124 @@
+terraform {
+  required_providers {
+    lxd = {
+      source  = "terraform-lxd/lxd"
+      version = "2.0.0"
+    }
+    ssh = {
+      source = "loafoe/ssh"
+      version = "2.7.0"
+    }
+  }
+}
+
+provider "lxd" {
+}
+provider ssh {
+  # Configuration options
+}
+locals {
+  cloud_init = <<EOT
```

Comment:
Can't the MicroCloud preseed file also be included in the cloud init? You can then use [`runcmd`](https://cloudinit.readthedocs.io/en/latest/reference/modules.html#runcmd) to execute the `microcloud init --preseed` command.

This way you won't need either SSH or the provisioner (and a key pair) to connect to the machine and to run the command.

Reply:
You could still use the templating engine from Terraform to generate the `preseed.yml`.

Reply:
After https://github.com/canonical/microcloud/pull/383 is being merged, you would need to put the preseed file on all of the nodes anyway. 
Also a initiator/leader has to be picked as part of crafting the preseed file (see https://github.com/canonical/microcloud/pull/383/files#diff-14d3c025154c62cbd2816232e7d56a1d80c9389b77d2eebdac004c646af8783d for reference).

With cloud-init you would then run `microcloud preseed` on all nodes and MicroCloud will pick the selected initiator/leader automatically.